### DISCUSSION TITLE
How to mock nuxt-auth-sanctum

### DISCUSSION THREAD

**Discussion by martijnhalekor** [manchenkoff/nuxt-auth-sanctum#185] [Q&A] [answered]
Hello and first of all thanks for this wonderful Nuxt module! <3

I'm currently setting up an SPA in Nuxt and need to start adding unit tests to my components. After adding my first smoke test, I noticed some logs by the nuxt-auth-sanctum module:
`[error] [nuxt-auth-sanctum:csr] Unable to load user identity from API [GET] "api/user": [...]`

Which got me wondering, how do I go about mocking this auth module? Any tips or examples are greatly appreciated.

**Comment by manchenkoff** [comment, marked as answer]
Hey @martijnhalekor, this is a really good question! Thanks for bringing it up here. 

I see here several options:
1. Keep some test Laravel API up and running for tests (this is what I usually do while developing new features for the module)
2. Write server functionality for your fixture application and use its URL in your nuxt config. Technically, all requests will be handled by your app, not the actual Laravel instance. But it requires a server to be supported.
3. You can disable `sanctum.client.initialRequest` for tests, so the module won't bother you if you are not interested in authentication testing
